src/utils/loaders/getCityFeed.ts

```diff
@@ -1,8 +1,18 @@
-import axios from 'axios'
+import axios from 'axios';
+import { LoaderFunctionArgs } from 'react-router-dom';
 
-export async function getCityFeed(location: string) {
-  const token = process.env.API_KEY
-  // const url = process.env.API_URL_LOCAL
-  const response = await axios.get(`https://api.waqi.info/feed/${location}/?token=${token}`)
-  return response.data
-}
\ No newline at end of file
+export const getCityFeed = async (input: LoaderFunctionArgs | string) => {
+  const token = process.env.API_KEY;
+
+  let location: string;
+  if (typeof input === 'string') {
+    location = input;
+  } else {
+    const { request, params } = input;
+    const url = new URL(request.url);
+    location = params.location || url.searchParams.get('location') || 'defaultLocation';
+  }
+
+  const response = await axios.get(`https://api.waqi.info/feed/${location}/?token=${token}`);
```

Comment:
can you create your own http methods file , such as : GET, PUT, POST, DELETE. and use axis only in there and in a ages, you can use only your own HTTP methods, so later on, you can easily replace axios with something else.

Reply:
http method file created with axios CRUD